pyro/contrib/gp/models/model.py

```diff
@@ -64,11 +82,97 @@ def forward(self, *args, **kwargs):
         """
         raise NotImplementedError
 
-    def _check_Xnew_shape(self, Xnew, X):
+    def _check_Xnew_shape(self, Xnew):
         """
         Checks the correction of the shape of new data.
         """
-        if Xnew.dim() != X.dim():
+        if Xnew.dim() != self.X.dim():
             raise ValueError("Train data and test data should have the same number of dimensions.")
-        if Xnew.dim() == 2 and X.size(1) != Xnew.size(1):
+        if Xnew.dim() == 2 and self.X.shape[1] != Xnew.shape[1]:
             raise ValueError("Train data and test data should have the same feature sizes.")
+
+    def _conditional(self, Xnew, X, kernel, f_loc, f_scale_tril=None, Lff=None, full_cov=False):
```

Comment:
Could you add a unit test for `GPModel._conditional()` for both cases of `full_cov`? It's easier for us to review tests than to review math (and I trust unit tests more than I trust myself to catch errors in math :wink:)

Reply:
Yup